server/src/main/java/org/apache/iotdb/db/wal/recover/WALNodeRecoverTask.java

```diff
@@ -226,6 +226,9 @@ private void recoverTsFiles() {
                 WALFileUtils.walFilenameFilter(dir, name)
                     && WALFileUtils.parseVersionId(name) >= firstValidVersionId);
     if (walFiles == null) {
+      for (UnsealedTsFileRecoverPerformer recoverPerformer : memTableId2RecoverPerformer.values()) {
+        recoverPerformer.getRecoverListener().succeed();
+      }
```

Comment:
Thank you for the bug fix, but UnsealedTsFileRecoverPerformer.endRecovery should be called before return success.

```suggestion
      for (UnsealedTsFileRecoverPerformer recoverPerformer : memTableId2RecoverPerformer.values()) {
      try {
        recoverPerformer.endRecovery();
        recoverPerformer.getRecoverListener().succeed();
      } catch (Exception e) {
        recoverPerformer.getRecoverListener().fail(e);
      }
    }
```

Reply:
Fixed, with a little of refactors.

Reply:
LGTM 👍